source/dynamical_systems/src/PointAttractor.cpp

```diff
@@ -154,6 +142,11 @@ void PointAttractor<JointState>::validate_and_set_parameter(const std::shared_pt
   }
 }
 
+template<class S>
+S PointAttractor<S>::compute_dynamics(const S&) const {
+  throw exceptions::NotImplementedException("compute_dynamics is not implemented for this type of DS");
+}
+
 template<>
 CartesianState PointAttractor<CartesianState>::compute_dynamics(const CartesianState& state) const {
   if (this->attractor_->get_value().is_empty()) {
```

Comment:
I cannot comment on the correct line but below here, compute dynamics will fail if the attractor is with 3 joints and the state to be evaluated is with 4 joints because we lost all notion of compatibility now

Reply:
you are right. I wanted to add a check in the PointAttractor, the only one that actually can do a check. I have added it now.

Reply:
Okay but what you did is not working either because now it requires the two JointStates to have the same names. But the attractor joint state will have name "attractor", so you should only check if the joint names correspond

Reply:
`is_compatible` is checking for name? We need to refactor that one anyway. Does not really make sense